src/solver/simulation/include/antares/solver/simulation/simulation-runner.h

```diff
@@ -22,17 +21,33 @@
 
 #pragma once
 
-#include "antares/infoCollection/StudyInfoCollector.h"
-#include "antares/solver/misc/options.h"
-#include "antares/solver/simulation/ISimulationObserver.h"
-#include "antares/writer/i_writer.h"
+#include "antares/solver/simulation/adequacy.h"
+#include "antares/solver/simulation/economy.h"
+#include "antares/solver/simulation/solver.h"
 
 namespace Antares::Solver
 {
-void runSimulationInAdequacyMode(Antares::Data::Study& study,
-                                 const Settings& settings,
-                                 Benchmarking::DurationCollector& durationCollector,
-                                 IResultWriter& resultWriter,
-                                 Benchmarking::OptimizationInfo& info,
-                                 Simulation::ISimulationObserver& simulationObserver);
-}
+
+class SimulationRunner
+{
+public:
+    SimulationRunner(Antares::Data::Study& study,
+                     const Settings& settings,
+                     Benchmarking::DurationCollector& durationCollector,
+                     IResultWriter& resultWriter,
+                     Simulation::ISimulationObserver& simulationObserver);
+
+    Benchmarking::OptimizationInfo run();
+
+private:
+    template<class simulationType>
+    Benchmarking::OptimizationInfo runSimulation();
+
+    Antares::Data::Study& study_;
+    const Settings& settings_;
+    Benchmarking::DurationCollector& durationCollector_;
+    IResultWriter& resultWriter_;
+    Simulation::ISimulationObserver& simulationObserver_;
+};
```

Comment:
A class is not required here, a free function would be enough.

```suggestion
template<class simulationType>
Benchmarking::OptimizationInfo runSimulation(Antares::Data::Study& study,
                 const Settings& settings,
                 Benchmarking::DurationCollector& durationCollector,
                 IResultWriter& resultWriter,
                 Simulation::ISimulationObserver& simulationObserver);
```

Reply:
Yes I hesitated on that point.

Reply:
I agree.
Done.